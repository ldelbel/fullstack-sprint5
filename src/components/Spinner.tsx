import React, { ReactElement, useContext } from "react";
import LoadingContext from "../contexts/LoadingContext";
import { ILoadingContext } from "../types/interfaces";
import "./Spinner.css";

function Spinner(): ReactElement<React.FC> {
  const { isLoading } = useContext<ILoadingContext>(LoadingContext);

  return isLoading() ? <div className="loading"></div> : <div></div>;
}

export default Spinner;
