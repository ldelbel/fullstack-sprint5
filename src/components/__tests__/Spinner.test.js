import React from "react";
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Spinner from "../Spinner";

jest.mock("react", () => {
  const react = jest.requireActual("react");
  return {
    ...react,
    useContext: jest.fn(),
  };
});

describe("Spinner component", () => {
  describe("when loading state is true", () => {
    it("should render", () => {
      React.useContext.mockImplementation(() => {
        return { isLoading: () => true };
      });
      const { container } = render(<Spinner />);
      expect(container.firstChild).toHaveClass("loading");
    });

    it("should match snapshot", () => {
      React.useContext.mockImplementation(() => {
        return { isLoading: () => true };
      });
      const spinner = renderer.create(<Spinner />).toJSON();
      expect(spinner).toMatchSnapshot();
    });
  });

  describe("when loading state is false", () => {
    it("should render an empty div", () => {
      React.useContext.mockImplementation(() => {
        return { isLoading: () => false };
      });
      const { container } = render(<Spinner />);
      expect(container.firstChild).toBeEmptyDOMElement();
    });
  });
});
