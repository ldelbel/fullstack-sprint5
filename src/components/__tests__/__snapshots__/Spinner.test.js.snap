// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component when loading state is true should match snapshot 1`] = `
<div
  className="loading"
/>
`;
